src/main/java/bio/terra/cli/apps/LocalProcessCommandRunner.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.cli.apps;
+
+import bio.terra.cli.apps.utils.LocalProcessLauncher;
+import bio.terra.cli.context.GlobalContext;
+import bio.terra.cli.context.WorkspaceContext;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** This class runs client-side tools in a local process. */
+public class LocalProcessCommandRunner extends CommandRunner {
+  private static final Logger logger = LoggerFactory.getLogger(LocalProcessCommandRunner.class);
+
+  private LocalProcessLauncher localProcessLauncher = new LocalProcessLauncher();
+
+  public LocalProcessCommandRunner(GlobalContext globalContext, WorkspaceContext workspaceContext) {
+    super(globalContext, workspaceContext);
+  }
+
+  /**
+   * Run a tool command in a local process.
+   *
+   * <p>Some setup/cleanup commands for gcloud authentication and configuration will be run
+   * before/after the given command.
+   *
+   * @param command the command and arguments to execute
+   * @param envVars a mapping of environment variable names to values
+   */
+  protected void runToolCommandImpl(List<String> command, Map<String, String> envVars) {
+    List<String> bashCommands = new ArrayList<>();
+    bashCommands.add("gcloud config set project " + workspaceContext.getGoogleProject());
```

Comment:
`gcloud`, `gsutil`, and `bq` are installed together as part of the Cloud SDK, and while you can remove certain of those tools afterwards (e.g. `gcloud components remove bq`), my guess is that's uncommon.

`nextflow` does not require `gcloud`, just defining `GOOGLE_APPLICATION_CREDENTIALS`. However, bash scripts that are called by a Nextflow workflow may well use `gcloud` CLIs, so I think it's good to have a consistent setup regardless of what command is running (i.e. so that `terra gcloud` doesn't work differently than `gcloud` called via a `terra nextflow` script).